### ISSUE TITLE
Consider providing a mechanism for building command line utilities in the main app 

### ISSUE THREAD

**Issue by mmihic** [square/misk#89]
One of the nice features of dropwizard / the internal SC is the ability to register command line utilities inside of the main application class, so that you can do things like:

````
java -jar buckets.jar upload-protos -f /path/to/my/protos.zip -b MyBucket
````

instead of
````
java -cp buckets.jar com.squareup.buckets.cli.UploadProtos -f /path/to/my/protos.zip -b MyBucket
````

Might be good to provide a similar ability. Important to note that different commands will almost certainly require different sets of Guice modules (e.g. only the ``server start`` command would embed the ``JettyModule``, for instance).

**Comment by mmihic** [comment]
Particularly nice since with Kotlin people will need to remember which commands have a Kt suffix and which don't

**Comment by eeSeeGee** [comment]
Maybe a generic Kubernetes Job definition where what you actually run is stored in a ConfigMap? That way you can specify yaml like:
`
runfile.sh: |
#bin/bash
export MY_IMPORTANT_ENV=something important
java -jar buckets.jar do-stuff to something misc_files/payload.json
`
The only thing the Job would do is
A) map the runfile.sh config to $HOMEDIR/command/runfile.sh in the pod filesystem
B) Optionally map another configmap to the misc_files directory
C) run runfile.sh

This way running a predefined command is a matter of telling Kubernetes to run job X, and the developer really doesn't have to worry about Kubernetes things, they just have to make a couple configmap yaml files (which can also be done with tooling). Additionally k8s events will track runs of these jobs, and the Job pods can hang around after they've terminated if we need to figure out what they did wrong.

**Comment by mmihic** [comment]
Resolved in PR #91
